data/types/device/factory/factory.go

```diff
@@ -57,8 +62,19 @@ func NewDeviceDatum(parser structure.ObjectParser) data.Datum {
 		return dataTypesDeviceTimechange.New()
 	case dataTypesDeviceParameter.SubType:
 		return dataTypesDeviceParameter.New()
+	case dataTypesDeviceMode.ConfidentialMode:
+		if eventType != nil {
+			return dataTypesDeviceMode.New(dataTypesDeviceMode.ConfidentialMode)
+		} else {
```

Comment:
The "else" is not needed here.
Since there is a return

Reply:
done